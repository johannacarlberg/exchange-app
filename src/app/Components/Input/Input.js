import React from 'react';
import PropTypes from 'prop-types';
import {
  Input,
  Indicator,
  InputContainer,
  Container
} from './Input.styles';

const InputBox = ({ value, indicator, onChange }) => {
  const dynamicWidth = value ? `${value.length * 20}px` : '20px';
  return (
    <Container>
      {value && <Indicator>{indicator}</Indicator> }
      <InputContainer width={dynamicWidth}>
        <Input onChange={onChange} value={value} placeholder="0"/>
      </InputContainer>
    </Container>
  );
};

InputBox.defaultProps = {
  value: '',
};

InputBox.propTypes = {
  value: PropTypes.string,
  indicator: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default InputBox;

