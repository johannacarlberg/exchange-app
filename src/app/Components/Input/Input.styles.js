import styled from 'styled-components';
import { DARK_GRAY } from '../../../utils/styles.constants';

export const InputContainer = styled.div`
// TODO do this 
  &::before {
    content: '-';
  }
`;

export const Input = styled.input`
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  -o-appearance: none;
  border: none;
  text-align: right;
  background: none;
  font-size: 2rem;
  outline: none;
  width: 200px;
  color: ${DARK_GRAY};
`;

export const Indicator = styled.i`

`;
