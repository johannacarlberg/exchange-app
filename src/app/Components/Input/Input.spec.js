import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const onChange = jest.fn();
  const tree = renderer
    .create(<Input indicator="+" value='23' onChange={onChange}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
