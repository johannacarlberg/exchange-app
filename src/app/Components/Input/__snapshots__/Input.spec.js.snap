// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bwzfXH hkrPvb"
>
  <i
    className="sc-bxivhb cvITPI"
  >
    +
  </i>
  <input
    className="sc-htpNat ePpogx"
    onChange={[MockFunction]}
    placeholder="0"
    value={10}
  />
</div>
`;
